cpu/atmega_common/periph/gpio.c

```diff
@@ -59,6 +63,88 @@
 #endif
 
 static gpio_isr_ctx_t config[GPIO_EXT_INT_NUMOF];
+
+/**
+ * @brief detects ammount of possible PCINTs
+ */
+#if defined(MODULE_ATMEGA_PCINT) || defined(MODULE_ATMEGA_PCINT0) || defined(MODULE_ATMEGA_PCINT1) || defined(MODULE_ATMEGA_PCINT2) || defined(MODULE_ATMEGA_PCINT3)
+#include "atmega_pcint.h"
+
+#ifndef ATMEGA_PCINT_MAP_PCINT0
+#error Please define pin change interrupts in atmega_pcint.h
+#endif /* ATMEGA_PCINT_MAP_PCINT0 */
+
+/**
+ * @brief check which pcints should be enabled!
+ */
+#if defined(MODULE_ATMEGA_PCINT) || defined(MODULE_ATMEGA_PCINT0)
```

Comment:
You can have module `atmega_pcint` depend on `atmega_pcint0` to `atmega_pcint3` in order to simplify your life here

Reply:
I am using the ATMEGA_PCINT module to enable *all* pcints at once.

Reply:
Using a makefile you can add all the individual bank modules when `atmega_pcint` is used. Then you do not need to handle that in C code

Reply:
See comment for `Makefile.include` to see what I meant

Reply:
But there might be different numbers of banks in CPUs, have to check if this leads to any problems if PCINT3 is enabled but only 0-2 exist.

Reply:
True. But I think it is still easier to be handled in the makefile, e.g. using

```
ifneq (,$(filter atmega_pcint,$(USEMODULE))
  USEMODULE += atmega_pcint0 atmega_pcint1 atmega_pcint2
  ifeq (atmega1284p,$(CPU))
    USEMODULE += atmega_pcint3
  endif
endif
```

Reply:
Maybe it might also make sense to use the `FEATURES_PROVIDED` and `FEATURES_REQUIRED` variables to guard against misuse. (E.g. a user might manually add `USEMODULE += atmega_pcint3` on an ATmega328p.) If the CPUs add `atmega_pcint<NUM>` for each supported vector to `FEATURES_PROVIDED` and the modules `atmega_pcint<NUM>` add this to `FEATURES_REQUIRED`, the build system will print a human readable error message when someone tries to use pin change interrupts for a non-existent bank

Reply:
I personally don't like the solution from above. This adds a lot of code specific code to the CPU. In the end, this isn't a problem, because I solved it directly in the code when checking if the bank is enabled: I just check if the mapping exists, if so, the CPU has the bank. Problem solved :)

Reply:
Working on the FEATURES_PROVIDED/FEATURES_REQUIRED solution right now.

Reply:
OK, I'm convinced. But maybe it would improve debugging when instead of

```C 
#if defined(MODULE_ATMEGA_PCINT1) && defined(ATMEGA_PCINT_MAP_PCINT1)
...
#endif
```

this is used:


```C 
#ifdef MODULE_ATMEGA_PCINT1
#ifndef ATMEGA_PCINT_MAP_PCINT1
#error "Either mapping for pin change interrupt bank 1 is missing or not supported by the MCU"
#endif /* ATMEGA_PCINT_MAP_PCINT1 */
...

#endif /* MODULE_ATMEGA_PCINT1 */
...
#endif
```

(This would indeed require cpu-depend handling of `atmega_pcint`, but uses will now what went wrong when the manually added `atmega_pcint<NUM>`

Reply:
Okay ... had to fix the dependencies, but it seems to be working now!

Reply:
True, will add some error messages for that!